store/postgres/src/deployment_store.rs

```diff
@@ -1441,6 +1441,11 @@ impl DeploymentStore {
         .await
     }
 
+    pub(crate) fn exists(&self, id: Arc<Site>) -> Result<bool, StoreError> {
+        let conn = self.get_conn()?;
```

Comment:
@leoyvens i tried that initially wouldn't that mean i would have to make all the callers also async function to await on it?

Reply:
[fn create_deployment_internal](https://github.com/graphprotocol/graph-node/blob/a54409d11485b75e90d1d6ce240ff31bbe40fd04/store/postgres/src/subgraph_store.rs#L530) is not async but its callers can be made async ( cause callers all the way is async ). Should i make them all async in that case?

Reply:
I see now that this is a bigger problem since `fn create_deployment_internal` is doing many sync DB operations, but is called from an async context. Ok we can leave this `exists` as sync for now, and later find a solution for the whole function.